developer_docs/ansible_dev_setup.asciidoc

```diff
@@ -0,0 +1,138 @@
+== Prerequisites
+
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/foreman_dev_setup.asciidoc[Foreman devel setup]
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/smart_proxy_dev_setup.asciidoc[Smart Proxy devel setup]
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/remote_execution_dev_setup.asciidoc[Remote Execution devel setup]
+* Installed https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html[Ansible]
+* Running machine that can be connected to via SSH (key authentication is best)
+
+[[foreman-plugin]]
+== Foreman plugin
+
+Download https://github.com/theforeman/foreman_ansible[foreman_ansible] plugin
+[source, bash]
+....
+git clone https://github.com/theforeman/foreman_ansible.git
+....
+
+Enable `foreman-ansible` plugin
+[source, ruby]
+....
+# bundler.d/ansible.local.rb
+
+gem 'foreman_ansible', path: '/path/to/foreman_ansible'
+....
+
+Install plugin dependencies & migrate database
+[source, bash]
+....
+bundle install
+npm install
+
+bundle exec rails db:migrate
+....
+
+
+[[smart-proxy-plugin]]
+== Smart Proxy plugin
+Add plugin gems
+
+[source, bash]
+....
+# /path/to/smart-proxy/bundler.d/ansible.local.rb
+
+gem 'smart_proxy_ansible', git: 'https://github.com/theforeman/smart_proxy_ansible.git'
+....
+
+Install gems
+[source, bash]
+....
+bundle install
+....
+
+Enable feature
+
+[source, yaml]
+....
+# config/settings.d/ansible.yml
+---
+:enabled: true
+....
+
+Run all servicies
+[source, bash]
+....
+bundle exec foreman start
+bundle exec rake dynflow:executor
+
+bundle exec bin/smart-proxy
+# => [I] Successfully initialized 'ansible'
+....
+
+[[foreman-collection]]
+== Foreman collection
+
+`theforeman.foreman` collection contains Ansible Modules to manage Foreman and Katello installations.
+
+Install https://docs.ansible.com/ansible/latest/collections/theforeman/foreman/index.html[Foreman Collection]
+
+[source, bash]
+....
+ansible-galaxy collection install theforeman.foreman
+....
+
+Create symbolic link to folder where you have your collections
+
+[source, bash]
+....
+sudo ln -s /home/<your-account>/.ansible/collections/ansible_collections/ /usr/share/ansible/collections/ansible_collections
+....
+
+[[foreman-callback]]
+== Foreman callback
+This https://docs.ansible.com/ansible/latest/collections/theforeman/foreman/foreman_callback.html[callback] will report facts and task events to Foreman
+
+Configure callback in `/etc/ansible/ansible.cfg`
+[source, bash]
+....
+[defaults]
+callbacks_enabled = foreman
+callback_plugins = /usr/share/ansible/collections/ansible_collections/theforeman/foreman/plugins/callback
+
+[callback_foreman]
+url = 'http://localhost:3000'
+....
+
+Create simple playbook `ping_playbook.yml`
+
+[source, yaml]
+....
+---
+- name: Ping Playbook
+  hosts: all
+  become: yes
+  become_user: root
+  gather_facts: true
+  tasks:
+    - name: Show host's hostname
+      debug:
+        msg: "{{ ansible_ssh_host }}"
```

Comment:
Is this enough? I was always explicitly running the `setup` task

Reply:
TIL that there is `setup` task. Well I never needed before, so I guess it's enough